### PR TITLE
Fix: Remove duplicate error message on crash (fixes #8964)

### LINKED ISSUE #8964: When a crash occurs, the error message is printed twice
**Tell us about your environment**

* **ESLint Version:** 4.2.0
* **Node Version:** 8.1.4
* **npm Version:** 5.0.3

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
rules:
  no-restricted-syntax: [error, 'Invalid selector [[[[']
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
a;
```

**What did you expect to happen?**

The configuration isn't valid (it includes an invalid selector), so I expected ESLint to output an error message.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint outputted the same error message twice, once with the stack trace and once without:

<details>
<summary>Output</summary>

```
Syntax error in selector "Invalid selector [[[[" at position 18: Expected " ", "." or [^ [\],():#!=><~+.] but "[" found.
Error: Syntax error in selector "Invalid selector [[[[" at position 18: Expected " ", "." or [^ [\],():#!=><~+.] but "[" found.
    at tryParseSelector (/path/to/node_modules/eslint/lib/util/node-event-generator.js:163:19)
    at lodash.memoize.rawSelector (/path/to/node_modules/eslint/lib/util/node-event-generator.js:175:28)
    at memoized (/path/to/node_modules/lodash/lodash.js:10566:27)
    at NodeEventGenerator.eventNames.forEach.rawSelector (/path/to/node_modules/eslint/lib/util/node-event-generator.js:235:30)
    at Array.forEach (native)
    at new NodeEventGenerator (/path/to/node_modules/eslint/lib/util/node-event-generator.js:234:20)
    at Linter.verify (/path/to/node_modules/eslint/lib/linter.js:909:34)
    at processText (/path/to/node_modules/eslint/lib/cli-engine.js:203:31)
    at processFile (/path/to/node_modules/eslint/lib/cli-engine.js:245:18)
    at executeOnFile (/path/to/node_modules/eslint/lib/cli-engine.js:585:25)

```

</details>

This isn't specific to the `no-restricted-syntax` rule -- it seems to occur whenever any error is thrown from anywhere.

### PR DESCRIPTION
**Pull request by nzakas**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[x] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Removed the extra console output that included just the error message and updated the associated tests.

**Is there anything you'd like reviewers to focus on?**

No

### CODE REVIEW

File: tests/bin/eslint.js [hunk 1]
```diff
@@ -333,12 +333,28 @@ describe("bin/eslint.js", () => {
...
+            return Promise.all([exitCodeAssertion, outputAssertion]);
+        });
+
         it("prints the error message pointing to line of code", () => {
             const invalidConfig = `${__dirname}/../fixtures/bin/.eslintrc.yml`;
             const child = runESLint(["--no-ignore", invalidConfig]);
             const exitCodeAssertion = assertExitCode(child, 2);
             const outputAssertion = getOutput(child).then(output => {
-                const expectedSubstring = "Error: bad indentation of a mapping entry at line";
+                const expectedSubstring = ": bad indentation of a mapping entry at line";
```
Comment (nzakas) [inline tests/bin/eslint.js]: The format of this error message was different on my machine. It said "YAMLException:" instead of "Error:". I split the difference and left the colon in.

Comment (not-an-aardvark) [inline tests/bin/eslint.js]: Can you add an assertion here that the error message here also contains the location of the invalid config file? e.g.

```js
assert.include(output.stderr, path.resolve(invalidConfig));
```

I vaguely remember that when I was looking at this, I was having trouble because the message "Cannot read config file: /path/to/config" wasn't appearing when I just printed `error.stack`. From testing this out locally it doesn't seem to be a problem anymore, but it would be good to have an assertion for it in case that changes in the future or it depends on the Node version.

Comment (nzakas) [reply tests/bin/eslint.js]: Sure, I'll add that.

Comment (nzakas) [reply tests/bin/eslint.js]: When I added that line, the test fails. There is no mention of the filename in the stack. I'd suggest opening a separate issue to track that. Agree?

Comment (not-an-aardvark) [reply tests/bin/eslint.js]: To clarify, was the filename previously getting printed for you before this change? Currently, the output for me (on `master`, with Node v10.10.0) looks like this:

```
$ eslint --no-ignore tests/fixtures/bin/.eslintrc.yml Makefile.js
Cannot read config file: /Users/tkatz/code/eslint/tests/fixtures/bin/.eslintrc.yml
Error: bad indentation of a mapping entry at line 5, column 9:
      quotes: error
            ^
YAMLException: Cannot read config file: /Users/tkatz/code/eslint/tests/fixtures/bin/.eslintrc.yml
Error: bad indentation of a mapping entry at line 5, column 9:
      quotes: error
            ^
    at generateError (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:165:10)
    at throwError (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:171:9)
    at readBlockMapping (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:1080:7)
    at composeNode (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:1332:12)
    at readDocument (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:1492:3)
    at loadDocuments (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:1548:5)
    at load (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:1569:19)
    at Object.safeLoad (/Users/tkatz/code/eslint/node_modules/js-yaml/lib/js-yaml/loader.js:1591:10)
    at loadYAMLConfigFile (/Users/tkatz/code/eslint/lib/config/config-file.js:95:21)
    at loadConfigFile (/Users/tkatz/code/eslint/lib/config/config-file.js:236:22)
```

To me, it seems like it's important for debugging that we tell users which config file has a problem when we report an "invalid config" error. So if the removal of `console.error(err.message)`also has the effect of removing this debugging info, then it seems like it might be worth leaving in the duplicate message for now (so we can make sure users continue to see the debugging info), until we have a better way to make sure that info is presented. On the other hand, if the name of the config file is already not appearing for you on `master`, then I'm fine with merging this PR now since it would be a clear improvement over the current situation.

In other words, my preference order for the possible outputs is:

1. Print an error message once, with the location of the invalid config
1. Print an error message twice, with the location of the invalid config
1. Print an error message once, without the location of the invalid config
1. Print an error message twice, without the location of the invalid config.

So I'm fine with merging this PR if it's moving us from state (4) to state (3), but I have concerns about it if it's moving us from state (2) to state (3).

Comment (nzakas) [reply tests/bin/eslint.js]: Gotcha, thanks for explaining. You are correct that `error.message` contains the filename while `error.stack` does not, so removing `error.message` will result in the filename not being output to the console.

(Side note: IIRC, this type of discrepancy is why I initially included both in the CLI. I actually don't think it's a problem to show duplicate error messages to get both the message and the stack, so I'm happy to just close the issue without merging this PR.)
